condor.go

```diff
@@ -330,51 +358,31 @@ func (cl *CondorLauncher) stopHandler(condorPath, condorConfig string) func(d am
 
 		invID = stopRequest.InvocationID
 
-		log.Infof("Running condor_rm for %s", invID)
-		if condorRMOutput, err = ExecCondorRm(invID, condorPath, condorConfig); err != nil {
-			log.Errorf("%+v\n", errors.Wrapf(err, "failed to run 'condor_rm %s'", invID))
+		if err = cl.stopJob(invID, condorPath, condorConfig); err != nil {
 			rejectDelivery(d, requeueOnErr, fmt.Sprintf("failed to Reject StopRequest for %s", invID))
 		} else {
-			fauxJob := model.New(cl.cfg)
-			fauxJob.InvocationID = invID
-			update := &messaging.UpdateMessage{
-				Job:     fauxJob,
-				State:   messaging.FailedState,
-				Message: "Job was killed",
-			}
-			if err = cl.client.PublishJobUpdate(update); err != nil {
-				log.Errorf("%+v\n", errors.Wrap(err, "failed to publish job update for a stopped job"))
-			}
-			log.Infof("condor_rm output for job %s:\n%s", invID, condorRMOutput)
-
 			ackDelivery(d, fmt.Sprintf("failed to ACK StopRequest for %s", invID))
-
-			cl.client.DeleteQueue(messaging.StopQueueName(invID))
 		}
 	}
 }
 
-func killHeldJobs(client *messaging.Client, condorPath, condorConfig string) {
+func killHeldJobs(launcher *CondorLauncher, condorPath, condorConfig string) {
 	var (
 		err         error
 		cmdOutput   []byte
-		heldEntries []queueEntry
+		heldEntries []string
 	)
 	log.Infoln("Looking for jobs in the held state...")
-	if cmdOutput, err = ExecCondorQ(condorPath, condorConfig); err != nil {
+	if cmdOutput, err = ExecCondorQHeldIDs(condorPath, condorConfig); err != nil {
 		log.Errorf("%+v\n", errors.Wrap(err, "error running condor_q"))
 		return
 	}
-	heldEntries = heldQueueEntries(cmdOutput)
+	heldEntries = heldQueueInvocationIDs(cmdOutput)
 	log.Infof("There are %d jobs in the held state", len(heldEntries))
-	for _, entry := range heldEntries {
-		if entry.InvocationID != "" {
-			log.Infof("Sending stop request for invocation id %s", entry.InvocationID)
-			if err = client.SendStopRequest(
-				entry.InvocationID,
-				"admin",
-				"Job was in held state",
-			); err != nil {
+	for _, invocationID := range heldEntries {
+		if invocationID != "" {
+			log.Infof("Sending stop request for invocation id %s", invocationID)
+			if err = launcher.stopJob(invocationID, condorPath, condorConfig); err != nil {
 				log.Errorf("%+v\n", errors.Wrap(err, "error sending stop request"))
```

Comment:
I suppose this line is inaccurate with the changes, though that's of course minor